jenkinsci/tfs-plugin
# Improve construction of TeamFoundationServerScm

This branch consists only of refactors and conversions designed to simplify the addition of fields representing new settings to the `TeamFoundationServerScm` class.

Instead of adding parameters to the constructor(s) and propagating to all the tests, adding a setter annotated with `@DataBoundSetter` works just as well and is much less disruptive.  The approach is also clearly documented.

Manual testing
--------------

1. I edited a job that used the TFVC SCM and confirmed that the settings were still being saved in the job's `config.xml` file and that nothing broke as a result of these changes.
    1. I tested adding and removing values to the `cloakedPaths` field and queuing the job.
    2. I experimented with turning on and off `userUpdate` and then queuing the job.
    3. The lack of catastrophy with the two tests above (plus its unit tests) gave me confidence that `localPath` wasn't also broken.
2. I created a new job that used the TFVC SCM to make sure the "new" scenario still worked.  I was able to queue that job as well and see it fetch items from the TFVC repository.

Mission accomplished!
